File: src/main/java/at/fhhagenberg/sqe/api/FloorButtonService.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.entity.FloorButton;

import java.rmi.RemoteException;
import java.util.List;

public interface FloorButtonService {
    public FloorButton get(int elevatorNumber, int floorNumber) throws RemoteException;
    public List<FloorButton> getAll(int elevatorNumber, int totalNumberOfFloors) throws RemoteException;
}

File: src/main/kotlin/at/fhhagenberg/sqe/api/FloorButtonService.kt
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.entity.FloorButton
import java.rmi.RemoteException

interface FloorButtonService {
    @Throws(RemoteException::class)
    fun getAll(elevatorNumber: Int): List<FloorButton>

    @Throws(RemoteException::class)
    fun get(elevatorNumber: Int, floorNumber: Int): FloorButton?
}